<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VideoPlayer</class>
 <widget class="QMainWindow" name="VideoPlayer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1215</width>
    <height>680</height>
   </rect>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>152</red>
        <green>209</green>
        <blue>240</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>230</red>
        <green>243</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>40</green>
        <blue>84</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>152</red>
        <green>209</green>
        <blue>240</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>230</red>
        <green>243</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>40</green>
        <blue>84</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>152</red>
        <green>209</green>
        <blue>240</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>230</red>
        <green>243</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>120</red>
        <green>120</green>
        <blue>120</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
    <weight>75</weight>
    <italic>false</italic>
    <bold>true</bold>
    <underline>false</underline>
    <kerning>true</kerning>
   </font>
  </property>
  <property name="windowTitle">
   <string>Interactive Music Playground</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="VideoWidget" name="inputFrame">
    <property name="geometry">
     <rect>
      <x>478</x>
      <y>30</y>
      <width>261</width>
      <height>171</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="openVideoFileButton">
    <property name="geometry">
     <rect>
      <x>558</x>
      <y>220</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Start Camera</string>
    </property>
   </widget>
   <widget class="VideoWidget" name="processedFrame">
    <property name="geometry">
     <rect>
      <x>478</x>
      <y>270</y>
      <width>261</width>
      <height>171</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="colorComboBox">
    <property name="geometry">
     <rect>
      <x>152</x>
      <y>80</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Choose color...</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>RED</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>YELLOW</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>GREEN</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>BLUE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PURPLE</string>
     </property>
    </item>
   </widget>
   <widget class="QScrollBar" name="minHueScrollbar">
    <property name="geometry">
     <rect>
      <x>148</x>
      <y>160</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>179</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="minSatScrollbar">
    <property name="geometry">
     <rect>
      <x>148</x>
      <y>200</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="minValScrollbar">
    <property name="geometry">
     <rect>
      <x>148</x>
      <y>240</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="maxHueScrollbar">
    <property name="geometry">
     <rect>
      <x>143</x>
      <y>340</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>179</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="maxSatScrollbar">
    <property name="geometry">
     <rect>
      <x>143</x>
      <y>380</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="maxValScrollbar">
    <property name="geometry">
     <rect>
      <x>143</x>
      <y>420</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="MIN_VALUES">
    <property name="geometry">
     <rect>
      <x>172</x>
      <y>127</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>MINIMAL VALUES</string>
    </property>
   </widget>
   <widget class="QLabel" name="MAX_VALUES">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>295</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>MAXIMAL VALUES</string>
    </property>
   </widget>
   <widget class="QLabel" name="HUE">
    <property name="geometry">
     <rect>
      <x>316</x>
      <y>161</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Hue</string>
    </property>
   </widget>
   <widget class="QLabel" name="SATURATION">
    <property name="geometry">
     <rect>
      <x>315</x>
      <y>200</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Saturation</string>
    </property>
   </widget>
   <widget class="QLabel" name="VALUE">
    <property name="geometry">
     <rect>
      <x>317</x>
      <y>239</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Value</string>
    </property>
   </widget>
   <widget class="QLabel" name="minhue">
    <property name="geometry">
     <rect>
      <x>103</x>
      <y>161</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="minsat">
    <property name="geometry">
     <rect>
      <x>102</x>
      <y>200</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="minval">
    <property name="geometry">
     <rect>
      <x>102</x>
      <y>240</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="maxhue">
    <property name="geometry">
     <rect>
      <x>104</x>
      <y>340</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="maxsat">
    <property name="geometry">
     <rect>
      <x>104</x>
      <y>378</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="maxval">
    <property name="geometry">
     <rect>
      <x>104</x>
      <y>420</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="medianBlurScrollbar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>404</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="singleStep">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="closingScrollbar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>541</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="openingScrollbar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>470</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="medianblur">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>404</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="opening">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>470</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="closing">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>541</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>384</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Median Blur</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>450</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Opening</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>521</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Closing</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>901</x>
      <y>80</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Minimum Chip Area</string>
    </property>
   </widget>
   <widget class="QLabel" name="minchipsize">
    <property name="geometry">
     <rect>
      <x>1071</x>
      <y>112</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>100</string>
    </property>
   </widget>
   <widget class="QPushButton" name="playAudioButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>470</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Start Audio</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="detectionToleranceScrollbar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>192</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>300</number>
    </property>
    <property name="value">
     <number>30</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="detectionTolerance">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>193</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>30</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>901</x>
      <y>162</y>
      <width>201</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Detection Tolerance (Frames)</string>
    </property>
   </widget>
   <widget class="QLabel" name="HUE_2">
    <property name="geometry">
     <rect>
      <x>318</x>
      <y>341</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Hue</string>
    </property>
   </widget>
   <widget class="QLabel" name="SATURATION_2">
    <property name="geometry">
     <rect>
      <x>316</x>
      <y>380</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Saturation</string>
    </property>
   </widget>
   <widget class="QLabel" name="VALUE_2">
    <property name="geometry">
     <rect>
      <x>319</x>
      <y>420</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Value</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="pathTextfield">
    <property name="geometry">
     <rect>
      <x>102</x>
      <y>620</y>
      <width>261</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>182</x>
      <y>590</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Resource Path</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="disableVideoOutput">
    <property name="geometry">
     <rect>
      <x>532</x>
      <y>520</y>
      <width>160</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Disable Video Output</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="positionChangedToleranceScrollBar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>271</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>241</y>
      <width>231</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Position Changed Tolerance (Pixel)</string>
    </property>
   </widget>
   <widget class="QLabel" name="positionChangedTolerance">
    <property name="geometry">
     <rect>
      <x>1071</x>
      <y>273</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>5</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="colorBox">
    <property name="geometry">
     <rect>
      <x>77</x>
      <y>50</y>
      <width>311</width>
      <height>411</height>
     </rect>
    </property>
    <property name="title">
     <string>Color Configuration</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="detectionBox">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>50</y>
      <width>271</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>Detection Configuration</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="MorphBox">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>350</y>
      <width>271</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Morphology</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="audioBox">
    <property name="geometry">
     <rect>
      <x>81</x>
      <y>490</y>
      <width>311</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Audio Configuation</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="minChipSizeScrollbar">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>110</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="minimum">
     <number>10</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <zorder>detectionBox</zorder>
   <zorder>MorphBox</zorder>
   <zorder>audioBox</zorder>
   <zorder>colorBox</zorder>
   <zorder>inputFrame</zorder>
   <zorder>openVideoFileButton</zorder>
   <zorder>processedFrame</zorder>
   <zorder>colorComboBox</zorder>
   <zorder>minHueScrollbar</zorder>
   <zorder>minSatScrollbar</zorder>
   <zorder>minValScrollbar</zorder>
   <zorder>maxHueScrollbar</zorder>
   <zorder>maxSatScrollbar</zorder>
   <zorder>maxValScrollbar</zorder>
   <zorder>MIN_VALUES</zorder>
   <zorder>MAX_VALUES</zorder>
   <zorder>HUE</zorder>
   <zorder>SATURATION</zorder>
   <zorder>VALUE</zorder>
   <zorder>minhue</zorder>
   <zorder>minsat</zorder>
   <zorder>minval</zorder>
   <zorder>maxhue</zorder>
   <zorder>maxsat</zorder>
   <zorder>maxval</zorder>
   <zorder>medianBlurScrollbar</zorder>
   <zorder>closingScrollbar</zorder>
   <zorder>openingScrollbar</zorder>
   <zorder>medianblur</zorder>
   <zorder>opening</zorder>
   <zorder>closing</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>label_4</zorder>
   <zorder>minchipsize</zorder>
   <zorder>playAudioButton</zorder>
   <zorder>detectionToleranceScrollbar</zorder>
   <zorder>detectionTolerance</zorder>
   <zorder>label_5</zorder>
   <zorder>HUE_2</zorder>
   <zorder>SATURATION_2</zorder>
   <zorder>VALUE_2</zorder>
   <zorder>pathTextfield</zorder>
   <zorder>label_6</zorder>
   <zorder>disableVideoOutput</zorder>
   <zorder>positionChangedToleranceScrollBar</zorder>
   <zorder>label_8</zorder>
   <zorder>positionChangedTolerance</zorder>
   <zorder>minChipSizeScrollbar</zorder>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>VideoWidget</class>
   <extends>QLabel</extends>
   <header>videowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
